cli/tests/integration_tests.rs

```diff
@@ -91,45 +98,142 @@ fn send_mined_sapling_to_orchard() {
             .unwrap();
-        println!("Send status: {send_status}");
 
         regtest_manager.generate_n_blocks(2).unwrap();
         sleep(Duration::from_secs(2)).await;
 
         client.do_sync(true).await.unwrap();
         let balance = client.do_balance().await;
-        let transactions = client.do_list_transactions(false).await;
-        println!("{}", json::stringify_pretty(balance.clone(), 4));
-        println!("{}", json::stringify_pretty(transactions, 4));
         assert_eq!(balance["verified_orchard_balance"], 5000);
     });
-    // Proposed Test:
-    //#[test]
-    //fn two_zcashds_with_colliding_configs() {
-    // Expectations:
-    //   The children are terminated by the test run end.
-    // Setup:
-    //   Two different zcashds are configured to launch with 1 config-location
-    //todo!("let _regtest_manager = setup::basic_funded_zcashd_lwd_zingolib_connected();");
-    // Execution:
-    //   Launch the two "bonking" zcashd instances
-    // Assertions:
-    //   zcashd A is terminated
-    //   zcashd B is terminated
-    //   The incorrectly configured location is still present (and not checked in)
-    //   The test-or-scenario that caused this situation has failed/panicked.
-    //}
 }
+/// This uses a manual outdated version of two_clients_a_coinbase_backed, but with the
+/// advantage of starting client_b on a different server, thus testing the ability
+/// to change servers after boot
+#[test]
+fn note_selection_order() {
+    let (regtest_manager_1, _child_process_handler_1, client_1, runtime) =
+        setup::coinbasebacked_spendcapable();
+    let (_regtest_manager_2, child_process_handler_2, client_2, _) =
+        setup::coinbasebacked_spendcapable();
+    // We just want the second client, we don't want the zcashd or lightwalletd
+    drop(child_process_handler_2);
+
+    runtime.block_on(async {
+        sleep(Duration::from_secs(1)).await;
+        regtest_manager_1.generate_n_blocks(5).unwrap();
+        sleep(Duration::from_secs(1)).await;
+        client_1.do_sync(true).await.unwrap();
+
+        client_2.set_server(client_1.get_server().clone());
+        client_2.do_rescan().await.unwrap();
+        let address_of_2 = client_2.do_address().await["sapling_addresses"][0].clone();
+        for n in 1..=5 {
+            client_1
+                .do_send(vec![(
+                    &address_of_2.to_string(),
+                    n * 1000,
+                    Some(n.to_string()),
+                )])
+                .await
+                .unwrap();
+        }
+        regtest_manager_1.generate_n_blocks(5).unwrap();
+        sleep(Duration::from_secs(2)).await;
+        client_2.do_sync(true).await.unwrap();
+        let address_of_1 = client_1.do_address().await["sapling_addresses"][0].clone();
+        client_2
+            .do_send(vec![(
+                &address_of_1.to_string(),
+                5000,
+                Some("Sending back, should have 2 inputs".to_string()),
```

Comment:
5_000 is more than "session 2" has in any single note (all of which received at `address_of_2`.